core/src/main/java/io/confluent/rest/ApplicationServer.java

```diff
@@ -228,7 +228,10 @@ private void configureConnectors() {
     for (NamedURI listener : listeners) {
       if (listener.getUri().getScheme().equals("https")) {
         if (httpConfiguration.getCustomizer(SecureRequestCustomizer.class) == null) {
-          httpConfiguration.addCustomizer(new SecureRequestCustomizer());
+          SecureRequestCustomizer secureRequestCustomizer = new SecureRequestCustomizer();
+          // Explicitly making sure that SNI is checked against Host in HTTP request
+          assert secureRequestCustomizer.isSniHostCheck();
```

Comment:
Since I don't know this code base, this comment might not be relevant, but...

Can we change the `assert` to an `if` statement with some logging? IIRC, the message the `assert` keyword spits out is very terse. I would think it would be helpful to add some logging for the user to correct the situation. Or maybe that's not possible in this case?

Reply:
thanks for pointing out, assert can spit out an uneasy error message, what is best I think is I'll change it to `Preconditions.checkArgument` for which we can write a nicer error message.